tools/clang/lib/SPIRV/DeclResultIdMapper.cpp

```diff
@@ -313,10 +317,34 @@ uint32_t DeclResultIdMapper::createSpirvStageVar(StageVar *stageVar,
       break;
     }
   }
-  // According to DXIL spec, the Depth SV can only be used by PSOut.
+  // According to DXIL spec, the Depth{|GreaterEqual|LessEqual} SV can only be
+  // used by PSOut.
   case hlsl::Semantic::Kind::Depth:
+  case hlsl::Semantic::Kind::DepthGreaterEqual:
+  case hlsl::Semantic::Kind::DepthLessEqual: {
     stageVar->setIsSpirvBuiltin();
+    if (semanticKind == hlsl::Semantic::Kind::DepthGreaterEqual)
+      theBuilder.addExecutionMode(entryFunctionId,
+                                  spv::ExecutionMode::DepthGreater, {});
+    else if (semanticKind == hlsl::Semantic::Kind::DepthLessEqual)
+      theBuilder.addExecutionMode(entryFunctionId,
+                                  spv::ExecutionMode::DepthLess, {});
     return theBuilder.addStageBuiltinVar(type, sc, BuiltIn::FragDepth);
+  }
+  // According to DXIL spec, the IsFrontFace SV can only be used by GSOut and
+  // PSIn.
+  // According to Vulkan spec, the FrontFacing BuitIn can only be used in PSIn.
+  case hlsl::Semantic::Kind::IsFrontFace: {
+    switch (sigPointKind) {
```

Comment:
since there are at most 2 cases (PSIn and GSOut ?), I would just use `if` statement instead of `switch`.